content/deep-dive/rpc/_index.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Remote Procedure Calls (RPC)
+weight: 6
+---
+
+Communication between services occurs over remote procedure calls. RPCs happen
+all the time in distributed systems. To obtain a webpage, your browser has to make at
+least one RPC to this website.
+
+TiKV, as a distributed system involving a number of nodes, uses RPCs to
+communicate between nodes, as well as the Placement Driver and clients.
+
+As it turns out, exposing functionality as a remote interface isn't a trivial.
+Networks are unreliable, systems are diverse and evolving, a huge variety of
+languages and formats exist, and even things like encoding are hard!
+
+## On the shoulders of giants
+
+Over the past decades the field of computing has largely settled on a few common
+standards.
+
+### Network Protocols
+
+The vast majority of services work over the HTTP or HTTP/2 network protocols.
+This solves problems such as:
+
+* [URIs]
+* [Sessions]
+* [Status Codes] (e.g. 404 Not Found)
+* [Methods] (e.g. GET, POST, PUT)
+* [Headers] (e.g. Authorization, User-Agent)
+* [Pipelining]
+
+TiKV uses **HTTP/2**. HTTP/2 is more performant and capable than HTTP/1 for TiKV uses.
+
+### Interfacing
+
+With those abilities supported, there remains a need to work with structures of
+data. Commonly this ends up being an *interface description language* format
+like [Protocol Buffers] (protobufs), which TiKV uses. Unlike an *interchange*
+*format*, an *interface description language* allows for the definition of
+services and RPCs in addition to just data interchange. This solves the
+following problems:
+
+* [Encoding] (ASCII? UTF-8? UTF-16? WTF-8?)
+* Serialization/Deserialization format (Text-to-`struct` and vice versa)

```

Comment:
To-structural object? Not sure, but not all programming languages call that `struct`.

Reply:
I think we're taking a fairly Rust centric perspective, and it's a generic enough term.